public/app/services/settingsService.js

```diff
@@ -1,53 +1,261 @@
 angular.module('kibibitCodeEditor')
 
-.service('SettingsService', ['Fullscreen', function(Fullscreen) {
-  var vm = this;
-
-  /* init */
-  vm.settings = new Settings();
-
-  function Settings() {
-    var settings = this;
-    /* INIT */
-    var cursor = {row: '0', column: '0'};
-    var isFullscreen = false;
-    var currentUndoManager = undefined;
-    var currentEditor = undefined;
-
-    /* EXPOSE SIMPLE VARS */
-    // This are probably better off in state instead of settings. but they're here for now :-)
-    settings.cursor = cursor;
-    settings.currentUndoManager = currentUndoManager;
-    settings.currentEditor = currentEditor;
-
-    settings.__defineSetter__('isFullscreen', function(newValue) {
-
-      console.assert(isBoolean(newValue), {
-        'message': 'isFullscreen should only be a boolean, but was given some other type',
-        'currentValue': isFullscreen,
-        'newValue': newValue
-      });
+.service('SettingsService', [
+  'Fullscreen',
+  'CODE_EDITOR',
+  'TYPE_ERROR_MSGS',
+  function(
+    Fullscreen,
+    CODE_EDITOR,
+    TYPE_ERROR_MSGS) {
+
+    var vm = this;
+
+    /* init */
+    vm.settings = new Settings();
+
+    function Settings() {
+      var settings = this;
+      /* INIT */
+      var cursor = {row: '0', column: '0'};
+      var isFullscreen = false;
+      var currentUndoManager = undefined;
+      var currentEditor = undefined;
+      var editorSettings = new EditorSettings();
+
+      /* EXPOSE SIMPLE VARS */
+      // This are probably better off in state instead of settings. but they're here for now :-)
+      settings.cursor = cursor;
+      settings.currentUndoManager = currentUndoManager;
+      settings.currentEditor = currentEditor;
+      settings.modelist = CODE_EDITOR.MODE_LIST;
+      settings.themelist = CODE_EDITOR.THEME_LIST;
+      settings.editorSettings = editorSettings;
 
-      if (newValue !== currentFullscreenState()) {
-        if (newValue) {
-          Fullscreen.all();
-        } else {
-          Fullscreen.cancel();
+      settings.__defineSetter__('isFullscreen', function(newValue) {
+
+        console.assert(isBoolean(newValue), {
+          'message': TYPE_ERROR_MSGS('isFullscreen', 'boolean',
+            typeof newValue),
```

Comment:
yes! loved this `typeof` use! very readable :100:  :-P

nit: I think I like this formatting better:
```javascript
console.assert(isBoolean(newValue), {
  'message':
    TYPE_ERROR_MSGS('isFullscreen', 'boolean', typeof newValue),
  'currentValue': isFullscreen,
  'newValue': newValue
});
```